### PR TITLE
add harfbuzz/2.6.8

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **harfbuzz/2.6.8**

this is a port of https://github.com/bincrafters/conan-harfbuzz

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/harfbuzz/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    def requirements(self):
+        if self.options.with_freetype:
+            self.requires("freetype/2.10.1")
+        if self.options.with_icu:
+            self.requires("icu/64.2")
+        if self.options.with_glib:
+            self.requires("glib/2.65.0")
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/harfbuzz/all/conanfile.py:49]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

File: recipes/harfbuzz/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        compiler = str(self.settings.compiler)
+        if compiler in ("clang", "apple-clang"):
+            flags.append("-Wno-deprecated-declarations")
+        cmake.definitions["CMAKE_C_FLAGS"] = " ".join(flags)
+        cmake.definitions["CMAKE_CXX_FLAGS"] = cmake.definitions["CMAKE_C_FLAGS"]
+
+        return cmake
+
+    def _configure_cmake_macos(self, cmake):
+        if str(self.settings.os) in ["Macos", "iOS", "watchOS", "tvOS"]:
```
Comment (uilianries) [inline recipes/harfbuzz/all/conanfile.py:77]:
```suggestion
        if tools.is_apple_os(self.settings.os):
```
https://docs.conan.io/en/latest/reference/tools.html#tools-is-apple-os

File: recipes/harfbuzz/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        self._cmake = CMake(self)
+        self._cmake = self._configure_cmake_compiler_flags(self._cmake)
+        self._cmake = self._configure_cmake_macos(self._cmake)
+        self._cmake.definitions["HB_HAVE_FREETYPE"] = self.options.with_freetype
+        self._cmake.definitions["HB_BUILD_TESTS"] = False
+        self._cmake.definitions["HB_HAVE_ICU"] = self.options.with_icu
+        self._cmake.definitions["HB_HAVE_GLIB"] = self.options.with_glib
+
+        if self.options.with_icu:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
```
Comment (jgsogo) [inline recipes/harfbuzz/all/conanfile.py:93]: Given this C++ standard requirement with ICU, we should add a check in the `configure()` method to validate if the provided compiler support C++17.

Comment (ericLemanissier) [reply recipes/harfbuzz/all/conanfile.py]: I just tested on my machine: `conan create /mnt/c/pf/conan-center-index/recipes/harfbuzz/all/ harfbuzz/2.6.8@ -o harfbuzz:with_icu=True -s compiler.cppstd=11 -b missing` passes without error
the cmake definition was originally added in this commit https://github.com/bincrafters/conan-harfbuzz/pull/10/commits/4ebf77bb7a11892e5595f5c638cf54ee725a9fdb#diff-2b1d42f71f22b7ea0412d7602dec166fR87 but it looks it was not necessary

**Pull request merged**